pkg/bindings/kafka_sarama/producer_message_encoder.go

```diff
@@ -76,7 +76,7 @@ func WriteKafkaProducerMessage(ctx context.Context, m binding.Message, producerM
 		producerMessage.Key = sarama.StringEncoder(s)
 	}
 
-	eventMessage := binding.EventMessage(e)
+	eventMessage := binding.EventMessage(*e)
```

Comment:
This provokes a copy, you should do `(*binding.EventMessage)(e)` to just cast it as a pointer